pkg/cnbutils/copy_project.go

```diff
@@ -11,31 +13,60 @@ import (
 	ignore "github.com/sabhiram/go-gitignore"
 )
 
-func CopyProject(source, target string, include, exclude *ignore.GitIgnore, utils BuildUtils) error {
+func CopyProject(source, target string, include, exclude *ignore.GitIgnore, utils BuildUtils, follow bool) error {
 	sourceFiles, _ := utils.Glob(path.Join(source, "**"))
+	knwonSymlinks := []string{}
+
 	for _, sourceFile := range sourceFiles {
+		if !shouldProcess(sourceFile, knwonSymlinks) {
+			continue
+		}
+
 		relPath, err := filepath.Rel(source, sourceFile)
 		if err != nil {
 			log.SetErrorCategory(log.ErrorBuild)
 			return errors.Wrapf(err, "Calculating relative path for '%s' failed", sourceFile)
 		}
 		if !isIgnored(relPath, include, exclude) {
 			target := path.Join(target, strings.ReplaceAll(sourceFile, source, ""))
-			dir, err := utils.DirExists(sourceFile)
+			isDir, err := utils.DirExists(sourceFile)
 			if err != nil {
 				log.SetErrorCategory(log.ErrorBuild)
 				return errors.Wrapf(err, "Checking file info '%s' failed", target)
 			}
+			isSymlink, err := symlinkExists(sourceFile, utils)
+			if err != nil {
+				return err
+			}
 
-			if dir {
+			if isSymlink && !isDir {
```

Comment:
Could we move the `if` and the first `else if` together? It is the same code, right? Only the condition for `knownSymlinks`. But would it hurt to add it anyway?